xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,164 @@
+        content1 = file1.readlines()
+        content2 = file2.readlines()
+
+    diff = list(difflib.Differ().compare(content1, content2))
+    filtered_diff: List[str] = []
+    for line in diff:
+        if line.startswith("+ ") or line.startswith("- "):
+            filtered_diff.append(line)
+
+    return filtered_diff
+
+
+patterns_for_noisy_lines = [
+    "^The taco parse function took [0-9]+ milliseconds to run$",
+    "^The xml write function took [0-9]+ milliseconds to run$",
+    "^The protobuf read function took [0-9]+ milliseconds to run$",
+    "^The protobuf write function took [0-9]+ milliseconds to run$",
+    "^$"
+]
+
+
+def remove_noisy_lines(lines: List[str]) -> List[str]:
+    filtered_array = []
+    for line in lines:
+        match_found: bool = False
+        for pattern in patterns_for_noisy_lines:
+            if re.fullmatch(pattern, line):
+                match_found = True
+                break
+        if not match_found:
+            filtered_array.append(line)
+    return filtered_array
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="A test harness for evaluating the output of the xmlconverter program")
+    parser.add_argument("-v", "--verbose", help="Prints the results from xmlconverter in JSON format", action="store_true")
+    parser.add_argument("-s", "--skiptests", help="Run the program but skip the comparison tests", action="store_true")
+    args = parser.parse_args()
+
+    with open(json_file_path, 'r') as json_file:
+        data = json.load(json_file)
+
+    for attribute_data in data:
+        attribute_name: str = attribute_data["attribute_name"]
+        print(attribute_name)
+
+        input_dir_path = os.path.join(tests_cases_path, attribute_name, "inputs")
+        expected_output_dir_path = os.path.join(tests_cases_path, attribute_name, "expected_outputs")
+        output_dir_path = os.path.join(tests_cases_path, attribute_name, "outputs")
+
+        # Ensure that the test output directory is created
+        os.makedirs(output_dir_path, exist_ok=True)
+
+        for test in attribute_data["tests"]:
+            file_name: str = attribute_name + "_" + test["name"] + ".xml"
+            input_xml_path = os.path.join(input_dir_path, file_name)
+            output_xml_path = os.path.join(output_dir_path, file_name)
+            expected_output_xml_path = os.path.join(expected_output_dir_path, file_name)
+
+            result = run_xml_converter(input_xml=[input_xml_path], output_xml=[output_xml_path])
+
+            # Remove noisy lines
+            stdout: List[str] = remove_noisy_lines(result[0].split("\n"))
+            stderr: List[str] = remove_noisy_lines(result[1].split("\n"))
+            returncode: int = result[2]
+
+            # Prints the results rather than comparing them to a file
+            if args.verbose:
+                print(f"\033[94mTest {attribute_name}_{test['name']}\033[0m")
```

Comment:
Lets remove the color codes being printed out for now to reduce the number of changes. You can add them back in in a different PR. When you do you can set variables equal to them so you can more easily understand what this string says. You could also turn them into functions like `red_text('Test {attribute_name)_{test['name']')` then